homeassistant/components/lutron_caseta/light.py

```diff
@@ -48,37 +53,145 @@ async def async_setup_entry(
+        if hs_color is not None:
+            color = FullColorValue(hs_color[0], hs_color[1])
+        elif kelvin_color is not None:
+            color = WarmCoolColorValue(kelvin_color)
+
+        # if user is pressing on button nothing is set, so set brightness to 255
+        if color is None and brightness is None:
+            brightness = 255
+
+        await self._set_brightness(brightness, color, **kwargs)
 
     async def async_turn_off(self, **kwargs: Any) -> None:
         """Turn the light off."""
-        await self._set_brightness(0, **kwargs)
+        await self._set_brightness(0, None, **kwargs)
+
+    @property
+    def color_mode(self) -> ColorMode:
+        """Return the current color mode of the light."""
+        warm_dim = self._device.get("warm_dim", False)
+        # check if warm dim is set, if so return white mode
+        if warm_dim:
+            return ColorMode.WHITE
+
+        # check if color is set, if so return color mode
+        current_color = self._device.get("color")
+        if isinstance(current_color, WarmCoolColorValue):
+            return ColorMode.COLOR_TEMP
+
+        if isinstance(current_color, FullColorValue):
+            return ColorMode.HS
+
+        # otherwise return default brightness mode
+        return ColorMode.BRIGHTNESS
 
     @property
-    def is_on(self):
+    def is_on(self) -> bool:
         """Return true if device is on."""
         return self._device["current_state"] > 0
+
+    @property
+    def hs_color(self) -> tuple[float, float] | None:
+        """Return the current color of the light."""
+        current_color = self._device.get("color")
+
+        # if bulb is set to full spectrum, return the hue and saturation
+        if isinstance(current_color, FullColorValue):
+            return (current_color.hue, current_color.saturation)
+
+        return None
+
+    @property
+    def color_temp_kelvin(self) -> int | None:
+        """Return the CT color value in kelvin."""
+        current_color = self._device.get("color")
+
+        # if bulb is set to warm cool mode, return the kelvin value
+        if isinstance(current_color, WarmCoolColorValue):
+            return current_color.kelvin
+
+        return None
+
+    @property
+    def max_color_temp_kelvin(self) -> int:
+        """Return maximum supported color temperature."""
+        white_tune_range = self._device.get("white_tuning_range")
+        # Default to 10k if not found
+        if white_tune_range is None or "Max" not in white_tune_range:
+            return 10000
+
+        return white_tune_range.get("Max")
+
+    @property
+    def min_color_temp_kelvin(self) -> int:
+        """Return minimum supported color temperature."""
+        white_tune_range = self._device.get("white_tuning_range")
+        # Default to 1.4k if not found
+        if white_tune_range is None or "Min" not in white_tune_range:
+            return 1400
+
+        return white_tune_range.get("Min")
```

Comment:
Can these change at run time?

If not let's use the shorthand `_attr` here so we don't have to calculate these every time state is written